### PR TITLE
Refactor `AppBase#preload`

### PR DESCRIPTION
**Pull request by willeastcott**
Dump unnecessary `Progress` class and simplify preloading (load and error callbacks are identical for example).

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/app-base.js [hunk 1]
```diff
@@ -689,62 +673,34 @@ class AppBase extends EventHandler {
...
-            // for each asset
-            for (let i = 0; i < assets.length; i++) {
-                if (!assets[i].loaded) {
-                    assets[i].once('load', onAssetLoad);
-                    assets[i].once('error', onAssetError);
-
-                    this.assets.load(assets[i]);
-                } else {
-                    progress.inc();
-                    this.fire("preload:progress", progress.count / total);
```
Comment (mvaligursky) [inline src/framework/app-base.js:739]: should we keep this event line?

Comment (mvaligursky) [reply src/framework/app-base.js:739]: ah actually you move it, all good

**Review by mvaligursky — APPROVED** [review]
much nicer

**Review by slimbuck — APPROVED** [review]
golly so much better <3

**Pull request merged**